src/V3LinkLValue.cpp

```diff
@@ -96,6 +98,24 @@ class LinkLValueVisitor final : public VNVisitor {
             m_setStrengthSpecified = false;
             iterateAndNextNull(nodep->rhsp());
         }
+        if (m_inInitialStatic && m_inFunc) {
+            const bool rhsHasIO = nodep->rhsp() && nodep->rhsp()->exists([](AstNodeVarRef* refp) {
+                // Exclude module I/O referenced from a function/task.
+                return refp->varp() && refp->varp()->isIO()
+                       && refp->varp()->lifetime() != VLifetime::NONE;
+            });
```

Comment:
I think references to module variables (not just "IO") should work:

```verilog
module top;
    bit a;
    function static f;
        static bit b = a;
    endfunction
endmodule
```

Also references to parameters/enum values, generally anything constant should definitely work.

Please add test cases and make sure we don't error on this.

Can we error here if and only if the reference in the initialzier is to a signal defined inside the function?